src/Microsoft.Azure.WebJobs.ServiceBus/EventHubs/EventHubAsyncCollector.cs

```diff
@@ -18,16 +19,13 @@ public class EventHubAsyncCollector : IAsyncCollector<EventData>
     {
         private readonly EventHubClient _client;
 
-        private List<EventData> _list = new List<EventData>();
-
-        // total size of bytes in _list that we'll be sending in this batch. 
-        private int _currentByteSize = 0;
-
+        private readonly Dictionary<string, PartitionCollector> _partitions = new Dictionary<string, PartitionCollector>();
```

Comment:
I'd prefer to see ConcurrentDictionary - is there any compelling reason we're managing the locking ourselves?